state/src/generated_types.rs

```diff
@@ -0,0 +1,239 @@
+// Generated by dsl::codegen from IR
```

Comment:
this is great for code review purposes, but it should be `.gitignore`-d, right?

Reply:
(for the code review, I'd add a link to a gist to examine the result quickly, or have the unit test run by Github print it to standard output into a dedicated section, and point to that section from the respective GH action run)

Reply:
Not sure it should be in .gitignore, tbh. Because it's an actual code that will be running, at some point - yes, it will be generated + compiled and shipped, but for now - idunno

Reply:
Sure, we'll `.gitignore` stuff later once we have running and passing tests!